my-webxr-app/tests/CsvUtils.test.tsx

```diff
@@ -0,0 +1,125 @@
+  });
+
+  it('should throws an error when the store does not exist within database', async () => {
+    (openDB as jest.Mock).mockResolvedValueOnce({
+      objectStoreNames: {
+        contains: () => false,
+      },
+    });
+
+    await expect(validateDbAndStore('testDb', 'testStore')).rejects.toThrow('Store "testStore" does not exist in database "testDb"');
+  });
+
+  it('should not throw an error if the database and store exist', async () => {
+    (openDB as jest.Mock).mockResolvedValueOnce({
+      objectStoreNames: {
+        contains: () => true,
+      },
+    });
+
+    await expect(validateDbAndStore('testDb', 'testStore')).resolves.not.toThrow();
+  });
+});
+
+describe('handleParsedData functions', () => {
+  it('should handle parsed data correctly, clearing the store first and then putting data', async () => {
+    const results: Array<Array<string | number | null>> = [['value1', 'value2']];
+    const dbName = 'testDb';
+    const storeName = 'testStore';
+
+    const mockClear = jest.fn().mockResolvedValue(undefined);
+    const mockPut = jest.fn().mockResolvedValue(undefined);
+
+    (openDB as jest.Mock).mockResolvedValueOnce({
+      transaction: () => ({
+        objectStore: () => ({
+          clear: mockClear,
+          put: mockPut,
+        }),
+        done: Promise.resolve(),
+      }),
+    });
+
+    await handleParsedData(results, dbName, storeName, 0);
+
+    expect(mockClear).toHaveBeenCalled();
+    results.forEach((item, index) => {
+      expect(mockPut).toHaveBeenCalledWith(item, index);
+    });
+  });
+
+  it('should clear the store but no put is called due to empty array', async () => {
+    const items: Array<Array<string | number | null>> = [];
+    const dbName = 'testDb';
+    const storeName = 'testStore';
+
+    const mockClear = jest.fn().mockResolvedValue(undefined);
+    const mockPut = jest.fn().mockResolvedValue(undefined);
+
+    (openDB as jest.Mock).mockResolvedValueOnce({
+      transaction: () => ({
+        objectStore: () => ({
+          clear: mockClear,
+          put: mockPut,
+        }),
+        done: Promise.resolve(),
+      }),
+    });
+
+    await handleParsedData(items, dbName, storeName, 0);
+
+    expect(mockClear).toHaveBeenCalled();
+    expect(mockPut).not.toHaveBeenCalled();
+  });
+});
+
+// This test only work on Node version 20 or higher
+
+// describe('parseAndHandleLocalCsv function', () => {
+//     it('should call Papa.parse with correctly passed arguments', async () => {
+//         const file = new File([""], "filename.csv", { type: "text/csv" });
+//         const dbName = 'testDb';
+//         const storeName = 'testStore';
+//         const setMessage = jest.fn();
+//
+//         await parseAndHandleLocalCsv(file, dbName, storeName, setMessage);
+//
+//         expect(Papa.parse).toHaveBeenCalledWith(file, expect.objectContaining({
+//             dynamicTyping: true,
+//             complete: expect.any(Function),
+//         }));
+//     });
+// });
```

Comment:
Since `File` wasn't added to Node until v20, could you use a `ReadableStream` instead? I was looking into it here: https://www.npmjs.com/package/papaparse#papa-parse-for-node

Reply:
I will take a look. I was hopelessly mocking a File and it is painful. Thanks much

Reply:
maybe not, cuz I want to pass in a file, which eventually gets parsed in chunks. If I use readbleStream here I will get type errors unless I modify the parseAndHandleLocalCsv() signature to take a stream of string instead of a file.

What ya thinks?

Reply:
Could it take both types? If it's not feasible that's fine; I was just thinking about trying to make the test work for both versions.